packages/server/src/tests/header/headerSelection.ts

```diff
@@ -107,91 +91,65 @@ const supportersTest: HeaderTest = {
     ],
 };
 
-const supportersTestUS: HeaderTest = {
-    name: 'header-supporter__USEOY',
-    audience: 'AllExistingSupporters',
-    variants: [
-        {
-            name: 'control',
-            modulePathBuilder: usEOYHeader.endpointPathBuilder,
-            content: {
-                heading: 'Thank you for supporting us',
-                subheading: '',
-                primaryCta: {
-                    url:
-                        'https://support.theguardian.com/us/contribute?selected-contribution-type=ONE_OFF',
-                    text: 'Make an extra contribution',
-                },
-            },
-            mobileContent: {
-                heading: 'Thank you',
-                subheading: '',
-                primaryCta: {
-                    url:
-                        'https://support.theguardian.com/us/contribute?selected-contribution-type=ONE_OFF',
-                    text: 'Contribute again',
-                },
-            },
-        },
-    ],
-};
+const isNotReportedAsSupporter = ['AllNonSupporters'];
+
+// Again, exported for Jest testing
+export const selectBestTest = (
+    targeting: HeaderTargeting,
+    allTests: HeaderTest[],
+): HeaderTestSelection | null => {
+    const { showSupportMessaging, countryCode } = targeting;
 
-const usEoyPeriodStart = new Date(2021, 10, 22);
-const usEoyPeriodEnd = new Date(2022, 1, 1);
+    const countryGroupId = countryCodeToCountryGroupId(countryCode.toUpperCase());
 
-const isInUsEoyPeriod = (date: Date): boolean => {
-    return isAfter(date, usEoyPeriodStart) && isBefore(date, usEoyPeriodEnd);
-};
+    const selectedTest: HeaderTest | undefined = allTests.find(test => {
+        const { isOn, userCohort, locations } = test;
+
+        if (!isOn) {
+            return false;
+        }
+
+        if (showSupportMessaging && !isNotReportedAsSupporter.includes(userCohort)) {
```

Comment:
I think we should share the [function from the banner](https://github.com/guardian/support-dotcom-components/blob/11353321bb2a961d7fe4f71e6e62a899ec26d877/packages/server/src/tests/banners/bannerSelection.ts#L76) here as the logic should be the same.
How about moving the function to here? -
https://github.com/guardian/support-dotcom-components/blob/11353321bb2a961d7fe4f71e6e62a899ec26d877/packages/server/src/lib/targeting.ts

Reply:
Function moved, as suggested